From a17348fbce0785db0e428c32140380e1a69d13e4 Mon Sep 17 00:00:00 2001
From: Daniel Dragomir <daniel.dragomir@windriver.com>
Date: Fri, 13 Mar 2026 02:17:35 +0200
Subject: [PATCH] Remove host contamination from extra_sources and
 extra_include_dirs

It also avoid buildpaths qa issue:

WARNING: python3-lmdb-1.8.1-r0 do_package_qa: QA Issue:
File /usr/lib64/python3.12/site-packages/lmdb/_config.py
in package python3-lmdb contains reference to TMPDIR
File /usr/lib64/python3.12/site-packages/lmdb/__pycache__/_config.cpython-312.pyc
in package python3-lmdb contains reference to TMPDIR [buildpaths]

Upstream-Status: Inappropriate
---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 1aedbc3..3391b08 100644
--- a/setup.py
+++ b/setup.py
@@ -81,8 +81,8 @@ elif os.getenv('LMDB_PURE') is not None:
     patch_lmdb_source = False
 else:
     print('py-lmdb: Using bundled liblmdb with py-lmdb patches; override with LMDB_FORCE_SYSTEM=1 or LMDB_PURE=1.')
-    extra_sources = [os.path.join(os.path.dirname(__file__), 'build/lib/mdb.c'), os.path.join(os.path.dirname(__file__), 'build/lib/midl.c')]
-    extra_include_dirs += [os.path.join(os.path.dirname(__file__), 'build/lib'), os.path.join(os.path.dirname(__file__), 'lib/py-lmdb')]
+    extra_sources = ['build/lib/mdb.c', 'build/lib/midl.c']
+    extra_include_dirs += ['build/lib']
     extra_compile_args += ['-DHAVE_PATCHED_LMDB=1']
     libraries = []
 
-- 
2.39.5

